editor/plugins/curve_editor_plugin.cpp

```diff
@@ -629,145 +762,270 @@ struct CanvasItemPlotCurve {
-		draw_string(font, get_view_pos(Vector2(0.25, y)) + off, "0.25", HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
-		draw_string(font, get_view_pos(Vector2(0.5, y)) + off, "0.5", HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
-		draw_string(font, get_view_pos(Vector2(0.75, y)) + off, "0.75", HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
-		draw_string(font, get_view_pos(Vector2(1, y)) + off, "1.0", HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
+	for (int i = 0; i <= grid_steps.x; ++i) {
+		real_t x = i * step_size.x;
+		draw_string(font, get_view_pos(Vector2(x - step_size.x / 2, curve->get_min_value())) + Vector2(0, font_height - Math::round(2 * EDSCALE)), String::num(x, 2), HORIZONTAL_ALIGNMENT_CENTER, get_view_pos(Vector2(step_size.x, 0)).x, font_size, text_color);
 	}
 
-	{
-		// Y axis
-		float m0 = curve.get_min_value();
-		float m1 = 0.5 * (curve.get_min_value() + curve.get_max_value());
-		float m2 = curve.get_max_value();
-		Vector2 off(1, -1);
-		draw_string(font, get_view_pos(Vector2(0, m0)) + off, String::num(m0, 2), HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
-		draw_string(font, get_view_pos(Vector2(0, m1)) + off, String::num(m1, 2), HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
-		draw_string(font, get_view_pos(Vector2(0, m2)) + off, String::num(m2, 3), HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
+	for (int i = 0; i <= grid_steps.y; ++i) {
+		real_t y = curve->get_min_value() + i * step_size.y;
+		draw_string(font, get_view_pos(Vector2(0, y)) + Vector2(2, -2), String::num(y, 2), HORIZONTAL_ALIGNMENT_LEFT, -1, font_size, text_color);
 	}
 
-	// Draw tangents for current point
+	// Draw curve.
 
-	if (_selected_point >= 0) {
-		const Color tangent_color = get_theme_color(SNAME("accent_color"), SNAME("Editor"));
+	// We set an unusual transform so we can offset the curve before scaling it up.
+	// The scaling up ensures that the curve rendering doesn't break when we use a quad line to draw it.
+	// This is so the curve can be antialiased.
+	draw_set_transform_matrix(Transform2D(0, get_view_pos(Vector2(0, curve->get_min_value()))));
 
-		int i = _selected_point;
-		Vector2 pos = curve.get_point_position(i);
+	const Color line_color = get_theme_color(SNAME("font_color"), SNAME("Editor"));
+	const Color edge_line_color = get_theme_color(SNAME("font_color"), SNAME("Editor")) * Color(1, 1, 1, 0.75);
 
-		if (i != 0) {
-			Vector2 control_pos = get_tangent_view_pos(i, TANGENT_LEFT);
-			draw_line(get_view_pos(pos), control_pos, tangent_color, Math::round(EDSCALE));
-			draw_rect(Rect2(control_pos, Vector2(1, 1)).grow(Math::round(2 * _gizmo_handle_scale * EDSCALE)), tangent_color);
-		}
+	CanvasItemPlotCurve plot_func(*this, line_color, edge_line_color);
+	plot_curve_accurate(**curve, 2.f, (get_view_pos(Vector2(1, curve->get_max_value())) - get_view_pos(Vector2(0, curve->get_min_value()))) / Vector2(1, curve->get_range()), plot_func);
```

Comment:
The transform can't do the heavylifting here, because while it worked for line primitives, for line quads it applies scaling, and without uniform scaling it would create an abomination.

![image](https://github.com/godotengine/godot/assets/85438892/a86c54f4-7978-46dd-8812-cb2f24a4a9c8)

Reply:
Is the issue that the scaling can be different in the X and Y dimensions? Or does it go beyond that?

IIRC, part of the whole `plot_func` trick was so that you could just pass it stuff in world coordinates and have it draw in view coordinates. So perhaps that's just a trick that's outlived its usefulness, and what we should really be doing is converting world-points into view-points and drawing in view coordinates. This should also get us anti-aliasing for free, since now the line-width is also given in view-coordinates.

There's no denying that this works right now, but like I said, the code does feel much more involved and awkward than I think it needs to be (though admittedly I haven't fooled around with it so I might be totally wrong). So the question is whether:
1. This PR gets some changes done so that the code hopefully goes back to being a lil simpler
2. Accept this as is, and have code that's temporarily more complicated, with a follow up PR attempting to clear it back up
3. Revert the stuff that gets us anti-aliasing in the name of clean code

I'm happy to leave this decision up to you since you know your availability best and some of this really depends on that! :)

Reply:
Option 2 please, unless the code is used in other places. But I'm not sure how to clean this up, so the cleanup might not be on me-